sw/device/lib/dif/dif_sram_ctrl.h

```diff
@@ -0,0 +1,215 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_DIF_DIF_SRAM_CTRL_H_
+#define OPENTITAN_SW_DEVICE_LIB_DIF_DIF_SRAM_CTRL_H_
+
+/**
+ * @file
+ * @brief <a href="/hw/ip/sram_ctrl/doc/">SRAM Controller</a> Device Interface
+ * Functions
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_warn_unused_result.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+/**
+ * Hardware instantiation parameters for SRAM Controller.
+ *
+ * This struct describes information about the underlying hardware that is
+ * not determined until the hardware design is used as part of a top-level
+ * design.
+ */
+typedef struct dif_sram_ctrl_params {
+  /**
+   * The base address for the SRAM Controller hardware registers.
+   */
+  mmio_region_t base_addr;
+} dif_sram_ctrl_params_t;
+
+/**
+ * A handle to SRAM Controller.
+ *
+ * This type should be treated as opaque by users.
+ */
+typedef struct dif_sram_ctrl {
+  dif_sram_ctrl_params_t params;
+} dif_sram_ctrl_t;
+
+/**
+ * The result of a SRAM Controller operation.
+ */
+typedef enum dif_sram_ctrl_result {
+  /**
+   * Indicates that the operation succeeded.
+   */
+  kDifSramCtrlOk = 0,
+  /**
+   * Indicates some unspecified failure.
+   */
+  kDifSramCtrlError = 1,
+  /**
+   * Indicates that some parameter passed into a function failed a
+   * precondition.
+   *
+   * When this value is returned, no hardware operations occured.
```

Comment:
Nit: s/occured/occurred/

Reply:
Good spot. Turns out this was a mistake in the template so I've sent #5390 to address it.